numpy/lib/npyio.py

```diff
@@ -731,7 +733,7 @@ def _getconv(dtype):
 
     def floatconv(x):
         x.lower()
-        if b'0x' in x:
+        if '0x' in x:
             return float.fromhex(asstr(x))
```

Comment:
No need for the `asstr` here, since the above line already requires x to be `unicode` in py3, and py2 doesn't care either way.